internal/encode.go

```diff
@@ -18,7 +18,6 @@ func EncodeQuery(v url.Values) string {
 	for k := range v {
 		keys = append(keys, k)
 	}
-	sort.Strings(keys)
```

Comment:
should we use a custom sort here instead of no sorting?  i kinda liked it was stable rendering of the input.  eg, sort where `secret` is always first

Reply:
something like this?
```go
	slices.SortFunc(keys, func(a, b string) bool {
		if a == "secret" {
			return true
		}
		if b == "secret" {
			return false
		}
		return a < b
	})
```

Reply:
sure thing - my biggest motivation is to roll out a fix for everyone to enjoy!

i can give you an alternative so you have another option to pick from:

```go
	// Google Authenticator expects the secret to be first, so handle it separately
	keys := make([]string, 0, len(v) - 1)
	for k := range v {
		if k != "secret" {
			keys = append(keys, k)
		}
	}
	sort.Strings(keys)
	for _, k := range append([]string{"secret"}, keys...) {
```

or maybe a little more optimised:

```go
	// Google Authenticator expects the secret to be first, so handle it separately
	keys := make([]string, 0, len(v))
	keys = append(keys, "secret")
	for k := range v {
		if k != "secret" {
			keys = append(keys, k)
		}
	}
	sort.Strings(keys[1:])
	for _, k := range keys {
```

these solutions fail, though, if there' no secret at all